src/support/z_file_entry.erl

```diff
@@ -198,89 +201,76 @@ locate(timeout, State) ->
+%%% gen_statem events
 %%% ------------------------------------------------------------------------------------
-
-handle_sync_event(lookup, From, locate, State) ->
-    {next_state, locate, State#state{waiting=[From|State#state.waiting]}, 0};
-handle_sync_event(lookup, _From, stopping, State) ->
-    {next_state, stopping, lookup_file_info(State), ?STOP_TIMEOUT};
-handle_sync_event(lookup, From, StateName, State) ->
+handle_event({call, From}, lookup, StateName, State) ->
     case check_current(State) of
         {ok, State1} ->
             Reply = lookup_file_info(State1),
-            {reply, Reply, StateName, State1, timeout(StateName, State1#state.is_found)};
+            gen_statem:reply(From, Reply),
+            {next_state, StateName, State1, timeout(StateName, State1#state.is_found)};
         stale ->
             State1 = State#state{waiting=[From|State#state.waiting]},
             {next_state, locate, State1, 0}
     end;
-handle_sync_event(force_stale, _From, _StateName, State) ->
-    {reply, ok, locate, State, 0};
-handle_sync_event(Msg, _From, StateName, State) ->
-    lager:error("Unexpected sync-event ~p in state ~p", [Msg, StateName]),
-    {next_state, StateName, State, ?SERVING_TIMEOUT}.
-
-
-handle_event(stop, _StateName, State) ->
+handle_event({call, From}, force_stale, _StateName, State) ->
+    {next_state, locate, State, [{timeout, 0}, {reply, From, ok}]};
+handle_event({call, _From}, stop, _StateName, State) ->
     unreg(State),
     {next_state, stopping, State, ?STOP_TIMEOUT};
-handle_event(Msg, StateName, State) ->
-    lager:error("Unexpected event ~p in state ~p", [Msg, StateName]),
-    {next_state, StateName, State, timeout(StateName, State#state.is_found)}.
-
-handle_info({gzip, Ref, Data}, StateName, #state{gzipper=Ref} = State) ->
+handle_event(info, {gzip, Ref, Data}, StateName, #state{gzipper=Ref} = State) ->
     State1 = State#state{
         gzip_part=#part_data{data=Data},
         gzip_size=size(Data),
         gzipper=undefined
     },
     {next_state, StateName, State1, timeout(StateName, State1#state.is_found)};
-handle_info({'DOWN', MRef, process, _Pid, _Info}, StateName, State) ->
+handle_event(info, {'DOWN', MRef, process, _Pid, _Info}, StateName, State) ->
     case is_mref_part(MRef, State#state.parts) of
         true ->
             {next_state, locate, State, 0};
         false ->
             {next_state, StateName, State, timeout(StateName, State#state.is_found)}
     end;
-handle_info(Info, StateName, State) ->
-    lager:info("Unexpected info ~p in state ~p", [Info, StateName]),
+handle_event(EventType, EventContent, StateName, State) ->
+    lager:error("Unexpected event ~p ~p in state ~p", [EventType, EventContent, StateName]),
     {next_state, StateName, State, timeout(StateName, State#state.is_found)}.
 
 code_change(_OldVsn, StateName, State, _Extra) ->
-    {next_state, StateName, State, timeout(StateName, State#state.is_found)}.
+    {ok, StateName, State}.
```

Comment:
I’m not sure; couldn’t find this in the docs: http://erlang.org/doc/man/gen_statem.html. 

This callback is optional, and only required for release upgrade (which we currently don’t use, right?). So perhaps it’s best to just remove the function?